projects/laji/src/app/shared-modules/document-viewer/document-annotation/document-annotation.component.ts

```diff
@@ -333,9 +335,20 @@ export class DocumentAnnotationComponent implements AfterViewInit, OnChanges, On
       if (this.document.gatherings) {
         this.document.gatherings.forEach(gathering => {
           if (gathering.units) {
-            const i = gathering.units.find(unit => unit.unitId === this.highlight);
-            if (i) {
-              return this.unitExist = true;
+            if (!this.identifying) {
+              const i = gathering.units.find(unit => unit.unitId === this.highlight);
+              if (i) {
+                return this.unitExist = true;
+              }
+            } else {
+              gathering.units.forEach(element => {
+                (element.media || []).forEach(image => {
+                  const i = image.fullURL === this.highlight;
+                  if (i) {
+                    return this.imgExist = true;
+                  }
+                });
+              });
             }
           }
         });
```

Comment:
`this.unitExists` and `this.imgExists` could be combined into a single variable. How about `unitOrImgExists`?  Also, I think using `some` would be more understandable here instead of a `forEach` since you are basically searching for a single occurrence of a unit or an img. It'd also be more performant, since you can stop the search right when you find a unit or an img (returning `true` doesn't stop the `forEach` in plain JS, though in jQuery it does :) Could also have less indentations.

Proposal (replaces lines 335-355):

```
      this.unitOrImgExists = this.document.gatherings?.some(({units = []}) =>
          this.identifying
            ? units.some(element =>
              (element.media || []).some(image => image.fullURL === this.highlight)
            )
            : units.some(unit => unit.unitId === this.highlight)
      );
```

Reply:
Actually even shorter:

```
this.unitOrImgExists = this.document.gatherings?.some(({units}) =>
  (units || []).some(unit =>
    this.identifying
      ? (unit.media || []).some(image => image.fullURL === this.highlight)
      : unit.unitId === this.highlight
  )
);
```